src/decode.jl

```diff
@@ -21,36 +22,49 @@ julia> path = "test/data/fire.gif"
 julia> img = gif_decode(path)
 60×30×33 Array{RGB{N0f8},3} with eltype RGB{N0f8}
 """
-function gif_decode(filepath::AbstractString)
-    Error = Cint(0)
-    gif = LibGif.DGifOpenFileName(filepath, Ref(Error))
+function gif_decode(filepath::AbstractString; info = false)
+    error = Cint(0)
+    gif = LibGif.DGifOpenFileName(filepath, Ref(error))
 
     try
         gif == C_NULL && error("failed to open the gif file: null pointer")
-        slurpReturn = LibGif.DGifSlurp(gif)
+        slurp_return = LibGif.DGifSlurp(gif)
 
-        if (slurpReturn != LibGif.GIF_OK)
-            println("Failed to read .gif file")
-            return false
+        if (slurp_return == LibGif.GIF_ERROR)
+            error("failed to read .gif file")
         end
-        N = unsafe_load(gif)
-        components = unsafe_wrap(Array, N.SavedImages, N.ImageCount)
-        d = unsafe_load(N.SColorMap)
-        palette = unsafe_wrap(Array, d.Colors, d.ColorCount)
-        final = zeros(RGB{N0f8}, N.SHeight, N.SWidth, N.ImageCount)
-        for i = 1:N.ImageCount
-            #erro handling for this
-            img = unsafe_wrap(Array, components[i].RasterBits, N.SWidth * N.SHeight)
+
+        loaded_gif = unsafe_load(gif)
+
+        img_count = loaded_gif.ImageCount
+        components = unsafe_wrap(Array, loaded_gif.SavedImages, loaded_gif.ImageCount)
+
+        # Gif's are palette based and can have upto 256 colors in a image
+        colormap = unsafe_load(loaded_gif.SColorMap)
+        palette = unsafe_wrap(Array, colormap.Colors, colormap.ColorCount)
+
+        final = zeros(RGB{N0f8}, loaded_gif.SHeight, loaded_gif.SWidth, loaded_gif.ImageCount)
```

Comment:
Can we unconditionally assume the input bytes are `RGB{N0f8}`, will it be, e.g., `ARGB{N0f8}` or `RGBA{N0f8}`, or if possible, `Gray{N0f8}`?

Reply:
We can unconditionally assume the input bytes are RGB{N0f8} as the GifColorType can hold only RGB:
https://github.com/ashwani-rathee/GIFImages.jl/blob/194c01c50ff928dc97c290a13d01e2bb16187680/lib/LibGif.jl#L18
as we know Gif's are palette based(max `256 Colors`)and uses colormap which can only contains colors of type GifColorType.